### PR TITLE
Core/Scripts: fix Nightfall proc chance and reduce it for victims with level above 60.

### PR DESCRIPTION
**Pull request by MokiGitHub**
Adding a TrinityCore commit https://github.com/TrinityCore/TrinityCore/commit/eac9c1f0b992fd3622fd7dd864b2acc2781ec778

P.S. 
I'm just learning C++.
My level -1)

## CHANGES PROPOSED:
-  Procchange must be reduced for all characters above level 60.

## TESTS PERFORMED:
- Build without errors and tested in-game.

## HOW TO TEST THE CHANGES:
- Add item Nightfall(19169).
- And go damage mob's 61-80 level and look proc.

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1597581112860069800.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1597581112860069800');
+
+-- Nightfall
+UPDATE `item_template` SET `spellppmRate_1`=2.5, `ScriptName`='item_generic_limit_chance_above_60' WHERE `entry`=19169;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1597581112860069800.sql:4]:
```suggestion
UPDATE `item_template` SET `spellppmRate_1`=2.5, `ScriptName`='item_generic_limit_chance_above_60' WHERE `entry`=19169;

```

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1597581112860069800.sql:4]: Just for newline <3

Comment (MokiGitHub) [reply data/sql/updates/pending_db_world/rev_1597581112860069800.sql]: Done

**Comment by pklloveyou** [comment]
Tested

File: src/server/scripts/World/item_scripts.cpp [hunk 2]
```diff
@@ -234,6 +234,30 @@ class item_captured_frog : public ItemScript
...
+public:
+    item_generic_limit_chance_above_60() : ItemScript("item_generic_limit_chance_above_60") { }
+
+    bool OnCastItemCombatSpell(Player* /*player*/, Unit* victim, SpellInfo const* /*spellInfo*/, Item* /*item*/) override
+    {
+        // spell proc chance gets severely reduced on victims > 60 (formula unknown)
+        if (victim->getLevel() > 60)
+        {
+            // gives ~0.1% proc chance at lvl 70
+            float const lvlPenaltyFactor = 9.93f;
```
Comment (FrancescoBorzi) [inline src/server/scripts/World/item_scripts.cpp:250]: we should avoid magic numbers and wrap them inside constants or enums

**Comment by felwyn** [comment]
Fix is working! Thank you for all involved!

**Review by Helias — APPROVED** [review]

**Pull request merged**